HW 30/main.js

```diff
@@ -0,0 +1,154 @@
+// Спливання та перехоплення дозволяють нам реалізувати один з найпотужніших шаблонів обробки подій під назвою делегування подій.
+// Ідея в тому, що якщо у нас є багато елементів, які обробляються подібним чином, то замість того, щоб призначати обробник кожному з них,
+// ми ставимо один обробник на їхнього спільного предка.
+// У обробнику ми отримуємо event.target, щоб побачити, де насправді сталася подія і обробити її.
+
+let ul = document.querySelector('#ul');
+
+ul.addEventListener('mousedown', function (e) {
+    e.preventDefault();
+});
+
+ul.addEventListener('click', function (e) { 
+    if (e.target == this) {
+        return false;
+    };
+
+    if (e.shiftKey) {
+        chooseSelected(this.children, e.target, ul);
+    } 
+    else {
+        if (!e.ctrlKey) {
+            clearSelected(this.children);
+        };   
+        toggleSelected(e.target);
+    }; 
+});
+
+function clearSelected(elems) {
+    for (let elem of elems) {
+        elem.classList.remove('selected');
+    };
+};
+
+function chooseSelected(elems, target, e) { 
+    target.classList.toggle('selected'); 
+    let node = target.previousElementSibling  
+
+    while(node != null) { 
+        if (node.classList != 'selected') {
+            node.classList.toggle('selected');
+        }
+        else {
+            break
+        };
+        node = node.previousElementSibling;  
+    };   
+}; 
+
+function toggleSelected(target) { 
+    target.classList.toggle('selected'); 
+};
+ 
+
+class Menu {
+    constructor(elem) {
+        this._elem = elem; 
+        this.ul = document.querySelector('#ul'); 
+        elem.onclick = this.onClick.bind(this);  
+        this.words = [
+                      'rock', 'paper', 'scissors', 'megaerror', 'escapology',
+                      'brightwork', 'verkrampte', 'protectrix', 'nudibranch', 'grandchild',
+                      'newfangled', 'flugelhorn', 'mythologer', 'pluperfect', 'jellygraph',
+                      'quickthorn', 'rottweiler', 'technician', 'cowpuncher', 'middlebrow',
+                      'jackhammer', 'triphthong', 'wunderkind', 'dazzlement', 'jabberwock',
+                      'witchcraft', 'pawnbroker', 'thumbprint', 'motorcycle', 'cryptogram'
+                     ];
+    }
+
+    addfirst() { 
+        let li = document.createElement("li"); 
+        let randomWords = this.words[Math.floor(Math.random()*this.words.length)]  + ' '  + this.words[Math.floor(Math.random()*this.words.length)];
```

Comment:
`Math.floor(Math.random()*this.words.length` - do not repeat yourself. It should be moved to separate function